### PR TITLE
[FLINK-29631] Update Jackson-bom to 2.13.4.20221013

### PR DESCRIPTION
**Pull request by snuyanzin**
Update of jackson-bom because of CVE-2022-42003
this was fixed within https://github.com/FasterXML/jackson-databind/issues/3590

### CODE REVIEW

File: flink-shaded-jackson-parent/flink-shaded-jackson-2/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -8,7 +8,7 @@ This project bundles the following dependencies under the Apache Software Licens
 
 - com.fasterxml.jackson.core:jackson-annotations:2.13.4
 - com.fasterxml.jackson.core:jackson-core:2.13.4
-- com.fasterxml.jackson.core:jackson-databind:2.13.4
+- com.fasterxml.jackson.core:jackson-databind:2.13.4.2
```
Comment (MartijnVisser) [inline flink-shaded-jackson-parent/flink-shaded-jackson-2/src/main/resources/META-INF/NOTICE:11]: Is this really the correct version, looking at what you're referring to in the POM file?

Comment (snuyanzin) [reply flink-shaded-jackson-parent/flink-shaded-jackson-2/src/main/resources/META-INF/NOTICE:11]: Based on issue description in databind repo [1] 
> 2.13.4.2 micro-patch (jackson-bom 2.13.4.20221013). (NOTE: 2.13.4.1/2.13.4.20221012 have an issue that affects Gradle users)

[1] https://github.com/FasterXML/jackson-databind/issues/3590#issue-1362567066

Comment (snuyanzin) [reply flink-shaded-jackson-parent/flink-shaded-jackson-2/src/main/resources/META-INF/NOTICE:11]: also `mvn dependency:tree` for this branch gives
```
[INFO]    +- com.fasterxml.jackson.core:jackson-annotations:jar:2.13.4:compile
[INFO]    +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.4.2:compile
[INFO]    |  \- com.fasterxml.jackson.core:jackson-core:jar:2.13.4:compile
[INFO]    +- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.13.4:compile
[INFO]    |  \- org.yaml:snakeyaml:jar:1.31:compile
[INFO]    +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.13.4:compile

```

Comment (MartijnVisser) [reply flink-shaded-jackson-parent/flink-shaded-jackson-2/src/main/resources/META-INF/NOTICE:11]: How freaking annoying...

**Review by MartijnVisser — APPROVED** [review]
Thanks @snuyanzin

**Pull request merged**